common/src/abstractions/import.service.ts

```diff
@@ -9,6 +10,14 @@ export abstract class ImportService {
   featuredImportOptions: readonly ImportOption[];
   regularImportOptions: readonly ImportOption[];
   getImportOptions: () => ImportOption[];
-  import: (importer: Importer, fileContents: string, organizationId?: string) => Promise<Error>;
-  getImporter: (format: ImportType, organizationId: string, password?: string) => Importer;
+  import: (
+    importer: Importer,
+    fileContents: string,
+    organizationId?: string
+  ) => Promise<ImportError>;
+  getImporter: (
+    format: ImportType | "bitwardenpasswordprotected",
```

Comment:
Just noticed that the abstract class is importing the `ImportType` from the implementation, which seems backwards. i.e. the implementation is defining the abstraction. Do you think it's worth lifting the `featuredImportOptions` and `regularImportOptions` to their own file, or (probably a better idea) lifting their `id`s out to a real `importType` enum? It's not quite as elegant as your current `ImportType` solution but feels like it's separating things out a bit more.

I've seen a few cases lately where we've stuck enums or types at the top of their related services instead of in their own file and I'm not sure about it. I can see why it was done here but once it starts growing it can get messy.

Reply:
The reason it's here and not in the abstract file or anywhere else is because the only place we use the split version of the ImportType (`featuredImportOptions` and `regularImportOptions`) is directly in this service.

If we move them out we'll have to export them rather than have them be private. I didn't like them being exported from the abstraction, it didn't make sense to me.

I think I'm OK with a new file. I don't see a reason to extract to an enum since I'm able to generate the `ImportType` type from our import option arrays.